resources/design-document.md

```diff
@@ -4,188 +4,185 @@
+Clicking the button for an existing warehouse will lead to a page where you can update a warehouse's inventory, with buttons leading to pages for Inventory Item creation and shipping reports
 
-PopStock will also provide a web interface for users to manage their warehouses and inventory.  There will be a main page showing existing warehouses, with buttons to create a new one or modify existing inventory items. The inventory item page will have buttons linking to pages to create, update, or delete existing inventory items. Clicking the button for an existing warehouse will lead to a page where you can update a warehouse's inventory, run reports, or generate a new layout.  
-# 6. API
+# 5. API
 
-## 6.1. Public Models
+## 5.1. Public Models
 ```
-//WarehouseModel 
-String id;
-String name;
-int region;
-Map<ItemModel, Integer> inventory
-int size
+//WarehouseModel
+String userId 
+String wareHouseId
+String name
 ```
 
 ```
-//ItemModel
-String id;
-String name;
-int regionalDemand;
-double salesForecast
-double perPallet;
-double weight;
-double purchaseCost;
-double baseMargin;
-double rateOfReplenishment;
-String category
-String synergy
-boolean active
+//TransactionModel
+transactionId
+String warehouseId
+String shipmentId
+String itemId
+int count
+Date date
+String partnerId
+String transactionType
 ```
 
 ```
-//shippedModel
-String itemId // partition key, string
-String warehouseId, sort key, string
-Date date
-int count
+//InventoryItemModel
+String itemId
+String warehouseId
+String name
+String category
 ```
 
-## 6.2. Create Item Endpoint
-
-* Accepts POST requests to /items
+## 5.2. Create Inventory Item Endpoint
+* Accepts POST requests to /warehouses/:warehouseId/inventory
 * Accepts input data and user info to create a new item and returns an itemModel for the logged-in user
-* 
-## 6.3 Get Item Endpoint
-* Accepts GET requests to /items/id
-* Accepts an itemId and user info and returns the relevant itemModel
-
-## 6.4 Remove Item Endpoint
-* Accepts PUT requests to /items/id/remove
-* Accepts an itemID, sets active to false for the given item, and returns the updated itemModel for the 
-
-## 6.5 Update Item Endpoint
-* Accepts PUT requests to /items/id
-* Accepts an itemId, input data, and user info, updates the item data, and returns an itemModel
-
-## 6.6 Create Warehouse Endpoint
+![](CreateItemSequenceDiagram.png)
+
+## 5.3 Get All Inventory Items Endpoint
+* Accepts Get requests to /warehouses/:warehouseId/inventory
+* Accepts a warehouseId and user info and returns a lit of all inventoryItems for the warehouse
+  *if the warehouseId is not found, will throw 'WarehouseNotFoundException'
+
+## 5.4 Remove Inventory Items Endpoint
+* Accepts Delete requests to /warehouses/:warehouseId/:inventoryItemId
+* Accepts a warehouseId, a list of inventoryItemIds, and user info, and deletes the inventory items from the inventory_items table
```

Comment:
Based on the description of what this endpoint would do, DELETE doesn't feel quite like the right HTTP method. When/how would this endpoint be called from the UI?